### ISSUE TITLE
Improve and clean client-server synchronization

### ISSUE THREAD

**Issue by clement-moulin-frier** [flowersteam/vivarium#2]
(I'll do this)

Currently there are several mechanisms in place:

- through `SimulatorController._start_timer()` in combination with `SimulatorClient.get_change_time()`
    - Currently this is implemented but no longer as effect, because `Simulator_server._change_time` is no longer modified
- through the `update_plot` callback in `panel_interface.py`

Current issues:
- `NotebookController` gets synchronized with the server only through explicit calls to `SimulatorClient.get_state()`
    - Issue: if one modifies the state from the panel interface, it won't be reflected in `NotebookController` unless there is an explicit call to `get_state`. 
    - A solution could be to revive the use of `change_time`. But then there would be a duplicated sync in `panel_interface`, both from `update_plot`and from `SimulatorController._start_timer()`. We could remove the latter, but then changes induced by `Simulator.update_fn()` won't be reflected (since they are notified in `SimulatorServer` --and probably shouldn't-- and therefore can't update `Simulator_server._change_time`).
    
So in summary, need to think more about the problem and potential solutions for improving this
